File: app/src/main/java/com/zest/android/home/OnHomeFragmentInteractionListener.java
package com.zest.android.home;

import com.zest.android.data.Recipe;

import java.util.List;

/**
 * To make an interaction between {@link HomeFragment} and {@link RecipeViewModel}
 */
public interface OnHomeFragmentInteractionListener {

    /**
     * to show the message of favorite
     *
     * @param message
     */
    void showMessage(int message);

    /**
     * to go to {@link com.zest.android.detail.DetailActivity}
     *
     * @param recipe
     */
    void gotoDetailPage(Recipe recipe);

    /**
     * To load all {@link Recipe}s
     *
     * @param recipes
     */
    void loadRecipes(List<Recipe> recipes);

    /**
     * To load favorite {@link Recipe} from DB
     *
     * @param recipe
     * @return
     */
    Recipe loadFavorite(Recipe recipe);

    /**
     * To remove favorite recipe from db by {@link Recipe#recipeId}
     *
     * @param recipe
     */
    void removeFavorite(Recipe recipe);

    /**
     * To insert favorite recipe in db
     *
     * @param recipe
     */
    void insertFavorite(Recipe recipe);


    /**
     * To show/hide {#EmptyView} by visibility
     *
     * @param visibility
     */
    void showEmptyView(boolean visibility);

}

File: app/src/main/java/com/zest/android/home/OnHomeFragmentInteractionListener.kt
package com.zest.android.home

import com.zest.android.data.Recipe

/**
 * To make an interaction between [HomeFragment] and [RecipeViewModel]
 */
interface OnHomeFragmentInteractionListener {

    /**
     * to show the message of favorite
     *
     * @param message
     */
    fun showMessage(message: Int)

    /**
     * to go to [com.zest.android.detail.DetailActivity]
     *
     * @param recipe
     */
    fun gotoDetailPage(recipe: Recipe)

    /**
     * To load all [Recipe]s
     *
     * @param recipes
     */
    fun loadRecipes(recipes: List<Recipe>)

    /**
     * To load favorite [Recipe] from DB
     *
     * @param recipe
     * @return
     */
    fun loadFavorite(recipe: Recipe): Recipe

    /**
     * To remove favorite recipe from db by [Recipe.recipeId]
     *
     * @param recipe
     */
    fun removeFavorite(recipe: Recipe)

    /**
     * To insert favorite recipe in db
     *
     * @param recipe
     */
    fun insertFavorite(recipe: Recipe)


    /**
     * To show/hide {#EmptyView} by visibility
     *
     * @param visibility
     */
    fun showEmptyView(visibility: Boolean)

}
